docs/core/compatibility/core-libraries/7.0/reflection-invoke-exceptions.md

```diff
@@ -0,0 +1,68 @@
+---
+title: ".NET 7 breaking change: Reflection invoke API exception changes"
+description: Learn about the .NET 7 breaking change in core .NET libraries where the exceptions thrown by reflection invoke APIs have changed.
+ms.date: 04/29/2022
+---
+# Changes to reflection invoke API exceptions
+
+The exceptions thrown when calling reflection invoke APIs have changed.
+
+## Previous behavior
+
+- Previously, when an invoked method that [returns a value by reference](../../../../csharp/programming-guide/classes-and-structs/ref-returns.md) returned `null`, a <xref:System.NullReferenceException> was thrown.
+
+- For constructors, the following exceptions were thrown:
+
+  - Transient exceptions, including <xref:System.OutOfMemoryException>.
+  - An <xref:System.OverflowException> when a negative value was passed for the length parameter of an array.
+
+- When `null` was passed for a [byref-like](xref:System.Type.IsByRefLike) parameter without the `ref` modifier (that is, passed by value), no exception was thrown, and the runtime substituted a default value for the null value.
+
+## New behavior
+
+Starting in .NET 7:
+
+- Instead of throwing the originating exception (including <xref:System.NullReferenceException> and <xref:System.OutOfMemoryException> mentioned in [Previous behavior](#previous-behavior)), <xref:System.Reflection.TargetInvocationException> is thrown in all cases after the initial `Invoke()` parameters are validated. The inner exception contains the originating exception.
```

Comment:
I was a bit confused about the NullReferenceException. Was it thrown by the invoked method previously, or did the method just return null and the Invoke API threw a NullReferenceException?